### DISCUSSION TITLE
`std::sync::RwLock` vs `tokio::sync::RwLock` for shared state

### DISCUSSION THREAD

**Discussion by jjangga0214** [tokio-rs/axum#2270] [General]
In [this shared state example](https://github.com/tokio-rs/axum/blob/dea36db400f27c025b646e5720b9a6784ea4db6e/examples/key-value-store/src/main.rs), `std::sync::RwLock` is used instead of `tokio::sync::RwLock`.

Is there a specific reason?

Thanks.

**Comment by jplatte** [comment]
See [Which kind of mutex should you use](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use) from tokio's docs. The same applies to `RwLock`s.
